monitor/src/main.rs

```diff
@@ -0,0 +1,167 @@
+            .short('o')
+            .long("repo_owner")
+            .required(false)
+            .help("The owner of the repo to watch"))
+        .arg(Arg::new("repo_name")
+            .short('n')
+            .long("repo_name")
+            .required(false)
+            .help("The name of the repo to watch"))
+        .arg(Arg::new("github_token")
+            .short('t')
+            .long("github_token")
+            .required(false)
+            .help("The GitHub token"))
+        .arg(Arg::new("actions_path")
+            .short('a')
+            .long("actions_path")
+            .required(false)
+            .help("The path to the actions file"))
+        .get_matches();
+
+    let config: Config = if let Some(config_path) = matches.get_one::<String>("config") {
+        println!("-- SealCI - Loading config from file: {:?}", config_path);
+        Config::from_file(config_path).expect("Failed to load config from file")
+    } else {
+        println!("-- SealCI - Loading config from CLI arguments");
+        let repo_name = matches.get_one::<String>("repo_name").expect("--repo_name argument is required");
+        Config {
+            configurations: vec![SingleConfig {
+                event: matches.get_one::<String>("event").expect("--event argument is required").to_string(),
+                repo_owner: matches.get_one::<String>("repo_owner").expect("--repo_owner argument is required").to_string(),
+                repo_name: repo_name.to_string(),
+                github_token: matches.get_one::<String>("github_token").expect("--github_token argument is required").to_string(),
+                actions_path: {
+                    let path = matches.get_one::<String>("actions_path").expect("--actions_path argument is required").to_string();
+                    Config::exists_actions_file(&path.clone(), &repo_name);
+                    path
+                }
+            }],
+            file_path: None,
+        }
+    };
+
+    println!("-- SealCI - Config loaded !");
+    println!("{:#?}", config);
+    config
+}
+
+async fn launch_api_server(configs: Arc<RwLock<Config>>) -> std::io::Result<()> {
+    let data = web::Data::new(AppState { configs: Arc::clone(&configs) });
+    HttpServer::new(move || {
+        App::new()
+            .app_data(data.clone())
+            .route("/configurations", web::get().to(get_configurations))
+            .route("/configurations", web::post().to(add_configuration))
+            .route("/configurations/{id}", web::get().to(get_configuration_by_id))
+            .route("/configurations/{id}", web::put().to(update_configuration))
+            .route("/configurations/{id}", web::delete().to(delete_configuration))
+            .route("/configurations/{id}/actions-file", web::get().to(get_actions_file))
+    })
+        .bind("127.0.0.1:8080")?
+        .run()
+        .await
+}
+
+async fn launch_github_listeners(configs: Arc<RwLock<Config>>, thread_handles: &mut Vec<JoinHandle<()>>) {
```

Comment:
You could put all your listeners in a seperate service. Something like, listeners_services.rs

Reply:
Not wanted for the MVP